// neword-core/dist/types/DocumentEntityType.js
export var DocumentEntityType;
(function (DocumentEntityType) {
    DocumentEntityType["INSTAGRAM_CAPTION"] = "INSTAGRAM_CAPTION";
    DocumentEntityType["INSTAGRAM_POST_IDEA"] = "INSTAGRAM_POST_IDEA";
    DocumentEntityType["INSTAGRAM_BIO_DESCRIPTION"] = "INSTAGRAM_BIO_DESCRIPTION";
    DocumentEntityType["INSTAGRAM_STORY_IDEA"] = "INSTAGRAM_STORY_IDEA";
    DocumentEntityType["INSTAGRAM_REEL_IDEA"] = "INSTAGRAM_REEL_IDEA";
    DocumentEntityType["INSTAGRAM_REEL_SCRIPT"] = "INSTAGRAM_REEL_SCRIPT";
    DocumentEntityType["FACEBOOK_POST_CAPTION"] = "FACEBOOK_POST_CAPTION";
    DocumentEntityType["FACEBOOK_POST_IDEA"] = "FACEBOOK_POST_IDEA";
    DocumentEntityType["FACEBOOK_PAGE_ABOUT"] = "FACEBOOK_PAGE_ABOUT";
    DocumentEntityType["LINKEDIN_POST_CAPTION"] = "LINKEDIN_POST_CAPTION";
    DocumentEntityType["LINKEDIN_POST_IDEA"] = "LINKEDIN_POST_IDEA";
    DocumentEntityType["LINKEDIN_PAGE_ABOUT"] = "LINKEDIN_PAGE_ABOUT";
    DocumentEntityType["TWITTER_TWIT"] = "TWITTER_TWIT";
    DocumentEntityType["THREADS_TWIT"] = "THREADS_TWIT";
    DocumentEntityType["EMAIL_CONTENT"] = "EMAIL_CONTENT";
    DocumentEntityType["EMAIL_IDEA"] = "EMAIL_IDEA";
    DocumentEntityType["SMS_CONTENT"] = "SMS_CONTENT";
    DocumentEntityType["SMS_IDEA"] = "SMS_IDEA";
    DocumentEntityType["PHONE_CALL_SCRIPT"] = "PHONE_CALL_SCRIPT";
    DocumentEntityType["GOOGLE_ADS_KEYWORDS"] = "GOOGLE_ADS_KEYWORDS";
    DocumentEntityType["GOOGLE_ADS_IDEA"] = "GOOGLE_ADS_IDEA";
    DocumentEntityType["WEBSITE_ABOUT"] = "WEBSITE_ABOUT";
    DocumentEntityType["WEBSITE_FAQ"] = "WEBSITE_FAQ";
    DocumentEntityType["WEBSITE_TESTEMONIAL"] = "WEBSITE_TESTEMONIAL";
    DocumentEntityType["WEBSITE_PRODUCT_DESCRIPTION"] = "WEBSITE_PRODUCT_DESCRIPTION";
})(DocumentEntityType || (DocumentEntityType = {}));
